### PR TITLE
Feature - Remove EXTERNAL_ADDRESS from default settings

### PR DESCRIPTION
**Pull request by dannypaz**
## Description
This PR removes EXTERNAL_ADDRESS from our default settings on the broker. The way this works, is that the broker will make connections from all engines to the relayer, so that the relayer doesn't need a static address back to the broker.

Example:
Broker wants to commit funds to the BTC/LTC market. Broker will make connections to the relayer through its BTC AND LTC engines and then will ask the relayer to open channels with it.

**Implementation Notes:**
For now, ive moved the action of `connectUser` into the `commit` action of the broker, however in a future change, we could probably move this to broker initialization AFTER the engines have validated as this action can be completed before committing to the Sparkswap Relayer

## Related PRs
List related PRs if applicable

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Link to Trello

### CODE REVIEW

**Comment by dannypaz** [comment]
[Remove default EXTERNAL_ADDRESS usage](https://trello.com/c/53yQg24K/1264-remove-default-externaladdress-usage)

File: broker-daemon/broker-rpc/wallet-service/commit.js [hunk 1]
```diff
@@ -96,11 +96,20 @@ async function commit ({ params, relayer, logger, engines, orderbooks }, { Empty
     logger.debug('Outbound channel already exists', { balance: maxOutboundBalance })
   }
 
-  const outboundPaymentChannelNetworkAddress = await engine.getPaymentChannelNetworkAddress()
-  const inboundPaymentChannelNetworkAddress = await inverseEngine.getPaymentChannelNetworkAddress()
+  logger.debug(`Connecting to the relayer with inverse engine: ${inverseSymbol}`)
+
+  // We want to make sure we are connected to the relayer on the inverseSymbol before
+  // we ask the relayer to open a channel to us. This is to make sure that the relayer
```
Comment (treygriffith) [inline broker-daemon/broker-rpc/wallet-service/commit.js]: I would specifically mention that it allows our node to not have a public IP and ports opened.

File: docker-compose.yml [hunk 2]
```diff
@@ -52,9 +52,9 @@ services:
       - RPC_USER=${BTC_RPC_USER}
       - RPC_PASS=${BTC_RPC_PASS}
       - RPC_HOST=${BTC_RPC_HOST:-}
+      - EXTERNAL_ADDRESS=${EXTERNAL_BTC_ADDRESS:-}
```
[the remark this answers is not in the document]
Comment (dannypaz) [reply docker-compose.yml]: @treygriffith that's true. let me see what I can do with the port

[the remark this answers is not in the document]
Comment (dannypaz) [reply docker-compose.yml]: @treygriffith there is honestly not a good way to do this. In both startup scripts for lnd https://github.com/sparkswap/lnd-engine/blob/master/docker/lnd/start-lnd-btc.sh#L22 we check if this address is blank. Either we do some weird string matching in that script (not ideal) or we just say "If the user wants to use an external address, they'll have to know that they must include the port"

[the remark this answers is not in the document]
Comment (dannypaz) [reply docker-compose.yml]: Basically, we cant set a default port here unless we split the two up. EXTERNAL_ADDRESS and EXTERNAL_PORT which might be another options @treygriffith

[the remark this answers is not in the document]
Comment (dannypaz) [reply docker-compose.yml]: https://github.com/sparkswap/lnd-engine/pull/185 PR is here

File: .env-testnet-sample [hunk 3]
```diff
@@ -50,15 +50,11 @@ LTC_RPC_PASS=sparkswapltc
 
 # External IP Addresses
 #
-# In order for your lightning nodes to be found by other users on the network
-# you must set an external address for each chain. If you are hosting all containers
-# on the same machine, then these will be the same.
+# This flag sets a static IP address for your LND instance. This address can either
+# be a TLD or an IP address. You will need to include the engine port for these addresses.
```
Comment (treygriffith) [inline .env-testnet-sample]: This statement is no longer true with the EXTERNAL_PORT changes

File: .env-testnet-sample [hunk 4]
```diff
@@ -50,15 +50,11 @@ LTC_RPC_PASS=sparkswapltc
 
 # External IP Addresses
 #
-# In order for your lightning nodes to be found by other users on the network
-# you must set an external address for each chain. If you are hosting all containers
-# on the same machine, then these will be the same.
+# This flag sets a static IP address for your LND instance. This address can either
```
Comment (treygriffith) [inline .env-testnet-sample]: We should note in the comments that this is optional.

Comment (dannypaz) [reply .env-testnet-sample]: @treygriffith I simply noted that the address is optional.

**Pull request merged**